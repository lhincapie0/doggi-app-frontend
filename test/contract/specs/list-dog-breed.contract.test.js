import { Matchers } from "@pact-foundation/pact";
import { getProvider } from "./init-pact";
import dogsApi from "../../../src/client/DogsApi";

const provider = getProvider();

describe("Given a Dog breed service", () => {
  beforeAll(async () => {
    console.log("Setting up provider...");
    await provider.setup();
  });

  describe("When a request to list dog breeds is made", () => {
    beforeAll(async () => {
      console.log("Setting up env for list dog breeds test...");

      await provider.addInteraction({
        uponReceiving: "A request to list a new dog breed",
        state: "list dog breeds",
        withRequest: {
          method: "GET",
          path: "/api/v1.0/dogbreeds/",
        },
        willRespondWith: {
          status: 200,
          body: Matchers.eachLike(
            {
              name: Matchers.string("Siberian Husky"),
              lifeExpectancy: Matchers.like(10),
              weight: Matchers.like(20),
              height: Matchers.like(40),
              country: Matchers.string("United Kingdom"),
              dogBreedNatures: Matchers.eachLike(
                {
                  id: Matchers.like(1),
                  name: Matchers.string('Intelligent'),
                },
                {
                  min: 1,
                }
              ),
              dogBreedColors: Matchers.eachLike(
                  {
                    id: Matchers.like(1),
                    name: Matchers.string('Red'),
                  },
                  {
                    min: 1,
                  }
              ),
            },
            { min: 1 }
          ),
        },
      });
    });

    it("Then it should return the right data", async () => {
      const listDogBreedsResponse = await dogsApi.getDogBreeds();
      expect(listDogBreedsResponse.data).toMatchSnapshot();
      provider.verify();
    });
  });

  afterAll(async () => {
    await provider.finalize();
  });
});
