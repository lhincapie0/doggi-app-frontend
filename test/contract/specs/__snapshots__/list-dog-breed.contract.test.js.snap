// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Dog breed service When a request to list dog breeds is made Then it should return the right data 1`] = `
Array [
  Object {
    "country": "United Kingdom",
    "dogBreedColors": Array [
      Object {
        "id": 1,
        "name": "Red",
      },
    ],
    "dogBreedNatures": Array [
      Object {
        "id": 1,
        "name": "Intelligent",
      },
    ],
    "height": 40,
    "lifeExpectancy": 10,
    "name": "Siberian Husky",
    "weight": 20,
  },
]
`;
